### ISSUE TITLE
ERROR: [WinError 32] The process cannot access the file because it is being used by another process

### ISSUE THREAD

**Issue by Mennaruuk** [Bluegrams/Vividl#28]
I have a couple of inquiries I'd like to bundle in one issue.

1. I'm getting this error. It's been frustrating as most of my downloads are failing. And this doesn't just include downloads where I'm converting to MP3 or WAV or any other format. By just downloading the original m4a (140), it can fail.
![Check error out later_censored(1)](https://user-images.githubusercontent.com/52135169/103234809-94d09700-490e-11eb-8a5c-885d190a44a8.jpg)

2. I have a bunch of videos I wanna download. The problem is, they all have the same title. Vividl doesn't name files in a sequential number order when it sees an apparent duplicate. Instead, after the first download, when the second video's download is finished, it simply replaces the first download's file. In essence, it is replacing the same file over and over again. I'm aware of the setting _Allow duplicate video entries_, but that's for pasting multiple copies of the same link into Vividl, not allowing duplicates as in that all the files with the same title will be saved.

**Comment by Mennaruuk** [comment]
Also, I'm getting this error, "WARNING: Unable to remove downloaded original file". This might be related to these videos with the same title. But I'm not sure.
![Screenshot 2020-12-28 145756_censored](https://user-images.githubusercontent.com/52135169/103240327-095f0200-491e-11eb-890f-d1e49436cff5.jpg)

**Comment by alxnull** [comment]
Just to make sure I got this right first: You're trying to download multiple videos with the same title in case 1 and case 2? The errors and warnings you get look like the program is trying to overwrite a currently downloading file with a new download of the same name.

The current behavior of using the same file name is actually partially intended to overwrite or skip the current download based on the "Overwrite existing files" setting. To solve your issue, I think it would make sense to add a new download mode which redownloads duplicates with an incrementing index, as you suggested. I'll look into this soon.

**Comment by Mennaruuk** [comment]
Yes, but I want to reiterate these videos aren't duplicates of each other. They simply share the same title. An example: say I have three videos from different channels. They're all titled YouTube. I'll call first video A, second video B, and third video C, in order. If I have concurrent downloads **off**, when A finishes downloading, B starts downloading and overwriting A. Then, when B finishes downloading, C starts downloading and overwriting B. If I have concurrent downloads **on**, it becomes a mess, because essentially all these files are overwriting each other. I tried unchecking _Overwrite existing files_ but that didn't do anything. (Examples of [two ](https://www.youtube.com/watch?v=AQEizp-VrVU)videos with [same title](https://www.youtube.com/watch?v=dYKxBKj29dI).)


So, to summarize,

1. Vividl should be downloading videos with the same title with some pattern naming rule so that files aren't overwriting each other. Just like how Firefox does it. So the first video is called YouTube, second can be called YouTube(1), and so on.
2. I'm getting WinError 32 error and another error saying that Vividl can't remove downloaded original file.
3. I also noticed it takes a long time for Vividl to load videos (that is, the time from pasting the links to the time the download button shows up). It can take up to a minute or two.

Thanks for your hard work!!!

**Comment by Mennaruuk** [comment]
I wanted to say one more thing: thank you for adding [haruhi-dl](https://git.sakamoto.pl/laudompat/haruhi-dl) support! It was a pleasant surprise to see this "error":

![image](https://user-images.githubusercontent.com/52135169/104977342-4ffbc580-59cd-11eb-8dcb-87dfffa69fe0.png)

I think it'd be fantastic if you can add some instructions for how to download media with haruhi-dl. What I did was point the youtube-dl path to where haruhi-dl is located (for me it's C:\Users\Computer\AppData\Local\Programs\Python\Python39\Scripts) then I clicked haruhi-dl to select it. Also, is there a way to have the user prioritize one over the other when both are installed?

Thank you!!

**Comment by alxnull** [comment]
> I wanted to say one more thing: thank you for adding haruhi-dl support! It was a pleasant surprise to see this "error":

You shouldn't thank me for this but the developers of haruhi-dl (I actually didn't even know about this tool) :)

In principle, it should be possible to use Vividl with all tools that have an interface compatible with youtube-dl. Great to see you could make it work with haruhi-dl. However, I can't guarantee that it continues to work as I'm not actively using or supporting it currently, but I can try to fix issues if you encounter and report any in the future.

(I also didn't get to publish a fix with the issues you described earlier since I'm quite busy at the moment. I hope I'll find some time to work on this again in the coming weeks.)

**Comment by Mennaruuk** [comment]
I definitely understand that the tool is primarily implemented with youtube-dl in mind, and that other programs like haruhi-dl are not guaranteed to work in the future. It's still a pleasant thing to know it's working right now, and I'm using it till it won't work anymore! And yes, their developers are super cool, and they are actively working to make this fork amazing!

And take your time please, I appreciate all the work you do!!! Stay awesome.

**Comment by alxnull** [comment]
Hey @Mennaruuk, I finally managed to release a new version: https://sourceforge.net/p/vividl/news/2021/06/new-features-in-vividl-v050/.
It has the download mode with incrementing indices for existing files we discussed. Go to `Settings -> Download -> Overwrite existing files` and select `Re-download (Rename)` to enable it. It currently only works for single video downloads and not for playlists though.

**Comment by Mennaruuk** [comment]
@alxnull **THANK YOU!** You are absolutely awesome. I can't wait for the update to drop on Chocolatey!

**Comment by alxnull** [comment]
@Mennaruuk Thanks so much for all your positive feedback! It really means a lot!
